cluster_utils/job.py

```diff
@@ -231,6 +239,7 @@ def singularity_wrap(
             "run" if singularity_settings.use_run else "exec",
             "--bind=%s" % ",".join(bind_dirs),
             "--pwd=%s" % cwd,
+            ", ".join(f"--env {name}={value}" for name, value in env_variables.items()),
```

Comment:
Why the comma for joining?  Shouldn't this resolve to something like `--env foo=1 --env bar=2` without commas in between?

Like above, it would be good to quote the value here to avoid issues with spaces.